rust/src/common.rs

```diff
@@ -121,3 +122,17 @@ pub fn to_hex(input: &[u8]) -> String {
         |b| vec![char::from(CHARS[(b >>  4) as usize]), char::from(CHARS[(b & 0xf) as usize])]
     ).flatten().collect();
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_to_hex(output: *mut u8, out_len: usize, input: *const u8, in_len: usize) {
+    static CHARS: &'static [u8] = b"0123456789abcdef";
+    let islice = build_slice!(input, in_len);
+    let oslice = std::slice::from_raw_parts_mut(output, out_len);
+    for i in 0..in_len {
```

Comment:
would there be an advantage to moving this inner loop into its own function that would not be `unsafe`? In general we try to avoid have too much logic in `unsafe` funcs

Reply:
cc @jasonish

Reply:
ok